xterm/index.cgi

```diff
@@ -7,13 +7,72 @@
 # XXX Virtualmin integration?
 
 require './xterm-lib.pl';
+&ReadParse();
 
+# Get Webmin current version for links serial
+my $wver = &get_webmin_version();
+$wver =~ s/\.//;
+
+# Build Xterm dependency links
+my $termlinks = 
+	{ 'css' => ['xterm.css?$wver'],
+	  'js'  => ['xterm.js?$wver',
+	  'xterm-addon-attach.js?$wver',
+	  'xterm-addon-fit.js?$wver'] };
+
+# Print header
 &ui_print_header(undef, $text{'index_title'}, "", undef, 1, 1, 0, undef,
-		 "<link rel=stylesheet href=xterm.css>\n".
-		 "<script src=xterm.js></script>\n".
-		 "<script src=xterm-addon-attach.js></script>\n"
+		 "<link rel=stylesheet href=\"$termlinks->{'css'}[0]\">\n".
+		 "<script src=\"$termlinks->{'js'}[0]\"></script>\n".
+		 "<script src=\"$termlinks->{'js'}[1]\"></script>\n".
+		 "<script src=\"$termlinks->{'js'}[2]\"></script>\n"
 		);
 
+# Set column size depending on the browser window size
+my $screen_width = int($ENV{'HTTP_X_AGENT_WIDTH'}) || int($in{'w'});
+my $screen_height = int($ENV{'HTTP_X_AGENT_HEIGHT'}) || int($in{'h'});
+if (!$screen_width ||
+    !$screen_height) {
+	print "<script>location.href = location.pathname + '?w=' + window.innerWidth + '&h=' + window.innerHeight;</script>";
+	return;
+}
+
+# Set pixel to columns conversion
+my $cols_num_user = int($screen_width / (int($in{'f'}) || 9));
+
+# Set pixel to rows (lines) conversion
+my $rows_num_user = int($screen_height / (int($in{'l'}) || 18));
+
+# Adjust columns and rows offset
```

Comment:
Why this adjustment?

Reply:
This is the way to determine the size on the user's browser window and adjust the the terminal accordingly.

Reply:
Oh, right, you're asking about the adjustments. Well, this is because we have a static container for the terminal. Static container would have, let's say 400px width. But calculated width would depend as each tiny block inside the terminal takes 9x18 pixels of WxH by default. This is to accommodate those 9 pixels in width and 18 pixels in height. Cockpit's terminal is also using Xterm.js and it seems very well implemented. I will look into a better (more precise) ways to do it for us as well. It's good already as it is but not yet perfect.

Reply:
Yas, queen! 🥳  I figured, and will submit the change after full rework and final tests.